waspc/data/Generator/templates/react-app/src/auth/forms/internal/common/LoginSignupForm.tsx

```diff
@@ -97,40 +97,55 @@ const googleSignInUrl = `${config.apiUrl}{= googleSignInPath =}`
 const gitHubSignInUrl = `${config.apiUrl}{= gitHubSignInPath =}`
 {=/ isGitHubAuthEnabled =}
 
+{=!
+// Since we allow users to add additional fields to the signup form, we don't
+// know the exact shape of the form values. We are assuming that the form values
+// will be a flat object with string values.
+=}
+export type LoginSignupFormFields = {
+  [key: string]: string;
+}
+
 export const LoginSignupForm = ({
     state,
     socialButtonsDirection = 'horizontal',
+    additionalSignupFields,
 }: {
-    state: 'login' | 'signup',
-    socialButtonsDirection?: 'horizontal' | 'vertical';
+    state: 'login' | 'signup'
+    socialButtonsDirection?: 'horizontal' | 'vertical'
+    additionalSignupFields?: AdditionalSignupFields
 }) => {
   const {
     isLoading,
     setErrorMessage,
     setSuccessMessage,
     setIsLoading,
   } = useContext(AuthContext)
-  const cta = state === 'login' ? 'Log in' : 'Sign up';
+  const isLogin = state === 'login'
+  const cta = isLogin ? 'Log in' : 'Sign up';
   {=# isAnyPasswordBasedAuthEnabled =}
   const history = useHistory();
   const onErrorHandler = (error) => {
     setErrorMessage({ title: error.message, description: error.data?.data?.message })
   };
   {=/ isAnyPasswordBasedAuthEnabled =}
+  const hookForm = useForm<LoginSignupFormFields>()
```

Comment:
We have a new depedency called `react-hook-form` and it's an industry standard. It enables easier form values management, client side validation and it makes it easier to add additional fields.